compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala

```diff
@@ -1125,4 +1125,29 @@ class JSInteropTest extends DirectTest with TestHelpers {
     """
   }
 
+  @Test
+  def noDefaultConstructorArgsIfModuleIsJSNative: Unit = {
+    """
+    @js.native
+    object A extends js.Object
+    @ScalaJSDefined
+    class A(x: Int = 1) extends js.Object
```

Comment:
Consider declaring the class before the object, to be consistent with the other test below.